app/controllers/projects_controller.rb

```diff
@@ -1,66 +1,72 @@
 class ProjectsController < ApplicationController
   before_action :authenticate_request, except: :index
   before_action :find_lesson
-  before_action :find_project, only: %i(update destroy)
+  before_action :find_project, only: %i(update destroy report)
 
   authorize_resource only: %i(update destroy)
 
   def index
-    @projects = projects
+    @projects = all_projects.page(params[:page]).per(50)
     @course = CourseDecorator.new(@lesson.course)
+    @user = current_user
+    @project = @projects.where(user_id: current_user.id).first if current_user
   end
 
   def create
     @project = new_project(project_params)
-    @project.save
-    set_recent_submissions
+    @project_saved = @project.save
+    @projects = latest_projects
   end
 
   def update
-    @project.update(project_params)
+    @project_updated = @project.update(project_params)
+    @decorated_project = ProjectDecorator.new(@project)
+  end
+
+  def report
```

Comment:
To keep things restful I think we should create a reports controller with a create action for this.